doc/manual/source/store/drv.md

```diff
@@ -0,0 +1,284 @@
+
+[`system` configuration option]: @docroot@/command-ref/conf-file.md#conf-system
+
+### Referencing derivations
+
+Derivations are always referred to by the [store path] of the store object they are encoded to.
+See the [encoding section](#derivation-encoding) for more details on how this encoding works, and thus what exactly what store path we would end up with for a given derivations.
+
+The store path of the store object which encodes a derivation is often called a "derivation path" for brevity.
+
+### Placeholder
+
+TODO
+
+Two types:
+
+- Reference to own outputs
+
+- output derived paths (see below), corresponding to store paths we haven't yet realized.
+
+> N.B. Current method of creating hashes which we substitute for string fields should be seen as an artifact of the current "ATerm" serialization format.
+> In order to be more explicit, and avoid gotchas analogous to [SQL injection](https://en.wikipedia.org/wiki/SQL_injection),
+> we ought to consider switching two a different format where we explicitly use a syntax for a oncatentation of plain strings and placeholders written more explicitly.
+
+## Deriving path {#deriving-path}
+
+Deriving paths are a way to refer to [store objects][store object] that might not yet be [realised][realise].
+This is necessary because, in general and particularly for [content-addressed derivations][content-addressed derivation], the [store path] of an [output] is not known in advance.
+There are two forms:
+
+- *constant*: just a [store path]
+  It can be made [valid][validity] by copying it into the store: from the evaluator, command line interface or another st    ore.
+
+- *output*: a pair of a [store path] to a [derivation] and an [output] name.
+
+In pseudo code:
+
+```idris
+type OutputName = String
+
+data DerivingPath
+  = ConstantPath { path : StorePath }
+  | Output {
+      drvPath : StorePath,
+      output  : OutputName,
+    }
+```
+
+[deriving path]: #deriving-path
+
+[content-addressed derivation]: @docroot@/glossary.md#gloss-content-addressed-derivation
+[realise]: @docroot@/glossary.md#gloss-realise
+[store object]: @docroot@/store/store-object.md
+[store path]: @docroot@/store/store-path.md
+
+## Encoding
+
+### Derivation {#derivation-encoding}
+
+There are two formats, documented separately:
+
+- The legacy ["ATerm" format](@docroot@/protocols/derivation-aterm.md)
+
+- The modern [JSON format](@docroot@/protocols/json/derivation.md)
+
+Currently derivations are always serialized to store objects using the "ATerm" format, but this is subject to change.
+
+Regardless of the format used, when serializing to store object, content-addressing is always used.
+In the common case the inputs to store objects are either:
+
+ - constant deriving paths for content-addressed source objects, which are "initial inputs" rather than the outputs of some other derivation (except in the case of bootstrap binaries).
```

Comment:
```suggestion
 - constant deriving paths for content-addressed source objects, which are "initial inputs" rather than the outputs of some other derivation
```

I think that happens outside the Nix world, so that's just confusing